ably-ios/ARTRest.m

```diff
@@ -200,10 +207,20 @@ - (void)time:(void(^)(NSDate *time, NSError *error))callback {
     return nil;
 }
 
-- (void)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback {
-    NSParameterAssert(query.limit < 1000);
-    NSParameterAssert([query.start compare:query.end] != NSOrderedDescending);
-    
+- (BOOL)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback error:(NSError **)errorPtr {
+    if (query.limit > 1000 && errorPtr) {
```

Comment:
So, it is not possible to call the super.dealloc:
<img width="700" alt="screen shot 2015-12-28 at 08 18 35" src="https://cloud.githubusercontent.com/assets/3541185/12015939/37099568-ad3c-11e5-83e5-93b7387dc942.png">

I checked with the compiler and it call first the child's dealloc and then the parent dealloc, so it is fine.